### PR TITLE
rdar://problem/26424505 Import as Member: QoI errors

### PR DESCRIPTION
**Pull request by jckarter**
Test updates necessary to keep up with the swift-clang changes in https://github.com/apple/swift-clang/pull/15 . This depends on that pull request getting merged into Clang's 3.0 branch simultaneously.

### CODE REVIEW

**Comment by jckarter** [comment]
@jrose-apple Do you mind reviewing these changes along with https://github.com/apple/swift-clang/pull/15 ?

**Comment by tkremenek** [comment]
@swift-ci test

**Comment by jckarter** [comment]
Failure is expected until apple/swift-clang#15 merges.

**Comment by tkremenek** [comment]
apple/swift-clang#15 has been merged.

**Comment by tkremenek** [comment]
@swift-ci smoke test os x platform

**Comment by tkremenek** [comment]
@swift-ci test linux

**Comment by tkremenek** [comment]
If Linux CI comes back good I'll merge.

File: test/IDE/Inputs/custom-modules/ImportAsMemberError.h [hunk 1]
```diff
@@ -19,15 +19,13 @@ typedef NSObject<ImportedProtocolBase> *ImportedProtocolBase_t;
...
+void mutateSomeStaticState(void)
     __attribute__((swift_name("ErrorProto.mutateSomeStaticState()"))); // ok
 void mutateSomeInstanceState(ErrorProto_t self) __attribute__((
     swift_name("ErrorProto.mutateSomeInstanceState(self:)"))); // error
 
 // Non-prototype declaration
 extern void IAMErrorStructHasPrototype(void)
     __attribute__((swift_name("ErrorStruct.hasPrototype()"))); // ok
-extern void IAMErrorStructNonPrototype()
-    __attribute__((swift_name("ErrorStruct.nonPrototype()"))); // error
```
Comment (jrose-apple) [inline test/IDE/Inputs/custom-modules/ImportAsMemberError.h]: This is pretty much the entire thing being tested, and now the comment is wrong. If there's a test on the Clang side I'm okay with removing it.

Comment (jckarter) [inline test/IDE/Inputs/custom-modules/ImportAsMemberError.h]: That was my thinking; now that we have better Clang-side validation and just throw the attr away if it's invalid, this wasn't really nonredundantly testing anything anymore.

**Pull request merged**